server/src/test/java/org/apache/druid/server/coordinator/duty/KillUnusedSegmentsTest.java

```diff
@@ -225,6 +225,154 @@ public void testKillWithMultipleDatasources()
     validateLastKillStateAndReset(DS2, null);
   }
 
+  /**
+   * Set up multiple datasources {@link #DS1}, {@link #DS2} and {@link #DS3} with unused segments with 2 kill task
+   * slots. Running the kill duty each time should pick at least one unique datasource in a round-robin manner.
+   */
```

Comment:
Style:
I generally prefer leaving one-line comments within the test method where relevant instead of javadocs (unless the test is too complicated). Ideally, the test name and the test code itself should be descriptive enough to clarify what is being verified.